.github/workflows/gcc13-compat.yml

```diff
@@ -0,0 +1,46 @@
+name: "MiNiFi-CPP verify gcc13 compatibility"
+on: [workflow_dispatch]
+jobs:
+  ubuntu_22_04:
+    name: "ubuntu-22.04-gcc-13"
+    runs-on: ubuntu-22.04
+    timeout-minutes: 120
+    steps:
+      - id: checkout
+        uses: actions/checkout@v3
+      - id: install_deps
+        run: |
+          sudo apt update
+          sudo apt install -y ccache libfl-dev libpcap-dev libboost-all-dev openjdk-8-jdk maven libusb-1.0-0-dev libpng-dev libgps-dev libsqliteodbc lua5.3 liblua5.3-dev flake8 parallel
+          echo "PATH=/usr/lib/ccache:$PATH" >> $GITHUB_ENV
+          echo -e "127.0.0.1\t$HOSTNAME" | sudo tee -a /etc/hosts > /dev/null
+      - name: build
+        run: |
+          eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"
+          brew install gcc@13
```

Comment:
You can use the Toolchain test PPA to install GCC 13: https://launchpad.net/~ubuntu-toolchain-r/+archive/ubuntu/test